src/main/java/com/guciowons/shoppingbasket/Basket/BasketService.java

```diff
@@ -0,0 +1,45 @@
+package com.guciowons.shoppingbasket.Basket;
+
+import com.guciowons.shoppingbasket.Product.ProductDao;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BasketService {
+    private Basket basket = new Basket();
+
+    private final ProductDao productDao;
+    private final BasketSummarizer basketSummarizer;
+
+    public BasketService(ProductDao productDao, BasketSummarizer basketSummarizer) {
+        this.productDao = productDao;
+        this.basketSummarizer = basketSummarizer;
+    }
+
+    public ResponseEntity<String> addProductToBasket(int id, int quantity) {
+        return productDao.findById(id)
+                .map(product -> {
+                    basket.addProduct(id, quantity);
+                    return new ResponseEntity<>("Done", HttpStatus.ACCEPTED);
+                }).orElseGet(() -> new ResponseEntity<>("No such product", HttpStatus.NOT_FOUND));
+    }
+
+    public ResponseEntity<String> removeProductFromBasket(int id, int quantity) {
+        return productDao.findById(id)
+                .map(product -> removeIfContains(id, quantity))
+                .orElseGet(() -> new ResponseEntity<>("No such product", HttpStatus.NOT_FOUND));
+    }
+
+    private ResponseEntity<String> removeIfContains(Integer productId, int quantity){
+        if(basket.getContent().containsKey(productId)){
```

Comment:
Jeżeli masz już obiekt `Basket`, który trzyma stan koszyka i zarządza dodawaniem i usuwaniem to cała logika dodawania i usuwania powinna być w nim -- tutaj logikę usuwanie rozrzucasz na dwie klasy. Tutaj sprawdzasz czy koszyk ma coś w contencie, a dopiero potem w nim coś usuwasz. To sprawdzenie też powinno być w klasie `Basket`.

Znowu -- poczytaj o SOLID i pierwszej zasadzie - Single Responsibility